src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -213,15 +200,21 @@ internal static ulong BigMul(uint a, ulong b, out ulong low)
         /// <returns>The high 64-bit of the product of the specified numbers.</returns>
         [CLSCompliant(false)]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe ulong BigMul(ulong a, ulong b, out ulong low)
+        public static ulong BigMul(ulong a, ulong b, out ulong low)
         {
+#if MONO // Multiply is not yet implemented in MONO
             if (Bmi2.X64.IsSupported)
             {
-                ulong tmp;
-                ulong high = Bmi2.X64.MultiplyNoFlags(a, b, &tmp);
-                low = tmp;
-                return high;
+                low = a * b;
+                return Bmi2.X64.MultiplyNoFlags(a, b);
```

Comment:
this change was not 100% intentional, the code is (at lest was) faster on normal dotnet/coreclr but I am not sure on mono.

I'll wait to switch back until I now if there are other changes that needs to be done